app/elastic/queries/person.py

```diff
@@ -17,16 +40,21 @@ def search_person(
         # Nom
         nom_person = search_params.dict().get(param_nom, "None")
         if nom_person:
+            # Remove stop words from the name
+            nom_person_filtered = remove_stop_words(nom_person)
+
             # match queries returns any document containing the search item,
             # even if it contains another item
-            for nom in nom_person.split(" "):
+            for nom in nom_person_filtered.split(" "):
```

Comment:
Why not apply the removing of stop words only to `nom` ?
```
nom_filtered = remove_stop_words(nom)
if not nom:
	continue;
```
This way 
remove_stop_words becomes :
```
def remove_stop_words():
	return '' if word.lower() in STOP_WORDS else word
```

Reply:
Also I think you should rewrite everything in Haskell

Reply:
I see two potential advantages to this approach :

* Memory Efficiency: for very long strings, processing one word at a time can reduce memory usage since we're not holding the entire string in memory while performing filtering

* If we expect that most of the input will consist of stop words, this approach can quickly skip over them without needing to generate a full list of filtered words upfront.

However, this approach can also introduce some overhead due to the increased number of function calls and iterations, which might negate some performance benefits. In my opinion, for short strings (like 5 to 6 words), the performance difference between the two is likely negligible, and the original method of processing the entire string at once remains efficient. At the end of the day, it seems to me as though both approches are equivalent.

What do you think ?